### DISCUSSION TITLE
Reading all messages from a websocket SplitStream

### DISCUSSION THREAD

**Discussion by kelvinlau20100** [tokio-rs/axum#2022] [Q&A] [answered]
Hey is there an easy way to read all messages that are currently in a SplitStream's stream e.g rcv.next().await until the stream is currently empty and then do something else? I cannot work out how to do this current

e.g 
```
let (snd, rcv) = sock.split();
// read from rcv until it's empty do something else for a few seconds 
// and then read from rcv till it's empty again
```

**Comment by davidpdrsn** [comment, marked as answer]
You can read from the stream with https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.next and if you wanna read for a few seconds you use combine it with https://docs.rs/tokio/latest/tokio/macro.select.html

If you wanna read the whole stream and wait you can use https://docs.rs/futures/latest/futures/prelude/stream/trait.StreamExt.html#method.collect
